src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+    
+    f = first(with_v_deg_1)
+    return -coeff(f, [v], [0])/coeff(f,[v],[1])
+end
+
+
+function small_reduce_one_step(Igens::Vector{<:RingElem}, phi::Oscar.MPolyAnyMap, 
+                               elim::Vector{<:RingElem}, fullyReduced::Bool)
+    Ivars = ideal_vars(Igens); 
+    length(Igens) == 0 && return (Igens, phi, elim, true)
+    for v in Ivars 
+        w = small_find_solution(v, Igens)
+        w isa String && continue
+        
+        push!(elim, v)
+        phi = update_hom(v, w, phi)
+        Igens = filter(x->x!=0, phi.(Igens))
+        return (Igens, phi, elim, false)
+    end
+    return (Igens, phi, elim, true)
+end
+
+
+function small_reduce_full_rec(Igens::Vector{<:RingElem}, 
+                               phi::Union{<:Oscar.MPolyAnyMap,Nothing} = nothing, 
+                               elim::Vector{<:RingElem} = Vector{RingElem}(), 
+                               fullyReduced=false)
+
+    if isnothing(phi)
+        R = parent(Igens[1])
+        phi=hom(R, R, a->a, gens(R))
+    end
+
+    output = small_reduce_one_step(Igens, phi, elim, fullyReduced)
+    output isa String && return "Not Realizable 0 in Semigroup"
+    
+    (Igens, phi, elim, fullyReduced) = output    
+    !fullyReduced && return small_reduce_full_rec(Igens, phi, elim, fullyReduced)
+    
+    R = domain(phi)
+    x = gens(R)    
+    cR = coefficient_ring(R)
+    
+    if length(elim) == length(x)
+        S = cR
+    else
+        S, y = polynomial_ring(cR, length(x) - length(elim))
+    end        
+    z=[]
+    j=1
+    for ele in x
+        if ele in elim
+            push!(z,0)
+        else
+            push!(z, y[j])
+            j+=1
+        end
+    end
+    phi2 = hom(R, S, a->a, z)
+    phif = phi*phi2
+
+    return phif
+end
+
+
+function small_reduce(MRS::MatroidRealizationSpace) 
+    
+    !MRS.realizable && return MRS
+    phi = small_reduce_full_rec(gens(MRS.defining_ideal))
+    new_ring = codomain(phi)
+    new_ideal = phi(MRS.defining_ideal)
+    
+    new_ideal_gens = filter!(x->!iszero(x), phi.(gens(MRS.defining_ideal)))
+    
+    if length(new_ideal_gens) == 0
+        new_ideal = ideal(new_ring, [new_ring(0)])
+    else
+        new_ideal = ideal(new_ideal_gens)
+    end
+
+    new_mat = phi.(collect(MRS.realization_matrix))
+    new_mat = matrix(new_ring, new_mat)
+    new_ineq = unique!(filter(x->!is_unit(x), phi.(MRS.inequations)))
+    new_ineq = gens_2_factors(new_ineq)
+    
+    return MatroidRealizationSpace(new_ideal, new_ineq, new_ring, new_mat, MRS.F, MRS.char, MRS.q)
+    
+end
+
+=#
+
+#####################
+# full reduction    #
+#####################
+
+# computes the coefficient of v in monomial m. 
+function coefficient_monomial(v::RingElem, m::RingElem)
+    isone(degree(m,v)) || return "The variable is not a degree 1 factor."
+    mf = factor(m)
+    mfdict = Dict(mf)
+    u = unit(mf)
+    not_v = [k^(mfdict[k]) for k in keys(mfdict) if k != v ]
```

Comment:
```suggestion
    u = unit(mf)
    not_v = [k^e for (k,e) in mf if k != v ]
```

Reply:
There are more places that needlessly convert a `Fac` object into a `Dict`, please check them as well.

Reply:
Fix here, and in the functions `clean`